src/Components/Components/src/BindConverter.cs

```diff
@@ -1213,6 +1213,27 @@ private static bool ConvertToNullableDateTimeOffsetCore(object? obj, CultureInfo
             return false;
         }
 
+        internal readonly static BindParser<Guid> ConvertToGuid = ConvertToGuidCore;
+
+        private static bool ConvertToGuidCore(object? obj, CultureInfo? culture, out Guid value)
+        {
+            var text = (string?)obj;
+            if (string.IsNullOrEmpty(text))
+            {
+                value = default;
+                return false;
+            }
+
+            if (!Guid.TryParse(text, out var converted))
+            {
+                value = default;
+                return false;
+            }
```

Comment:
I must be missing something here. How come you don't need a `try`/`catch` any more? I thought that adding the `try`/`catch` was the main reason for the change.

Does structuring the code this way somehow mean there isn't an unhandled exception any more? If you could clarify that would be great!

Reply:
Yes, because before the [`GuidConverter`](https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/GuidConverter.cs#L38-L45) would call the ctor which throws on invalid values. https://github.com/dotnet/aspnetcore/issues/33587#issuecomment-840525039
You can see it in the stack trace from here https://github.com/dotnet/aspnetcore/issues/33587#issue-922962519

---

Should there be another case for `Guid?`?

Reply:
That's correct.  Calling Guid.TryParse sets the ThrowStyle to None - https://github.com/dotnet/runtime/blob/5221db92ed579f8ffa56f1d8e84f07ba6d896077/src/libraries/System.Private.CoreLib/src/System/Guid.cs#L242
But calling the Guid ctor sets the ThrowStyle to All - https://github.com/dotnet/runtime/blob/5221db92ed579f8ffa56f1d8e84f07ba6d896077/src/libraries/System.Private.CoreLib/src/System/Guid.cs#L210

>Should there be another case for Guid??
Do you mean like the check for null/empty string?  The Guid class performs several checks for length and formatting, so that current check could be removed as well, but I left it since it would prevent having to call TryParse for the simplest invalid case.

Reply:
I meant for the `typeof(Guid?)` type, all the others do both checks like `typeof(DateTimeOffset)` and `typeof(DateTimeOffset?)`:
https://github.com/dotnet/aspnetcore/blob/a6e00f673d5da8188e85fd02159c51eac03adb09/src/Components/Components/src/BindConverter.cs#L1495-L1502

Reply:
You're right. I've just added another case for Nullable Guid.